lib/stats_reporter.rb

```diff
@@ -0,0 +1,98 @@
+# generates text for a report on storage root size and object status
+class StatsReporter
+
+  require 'text-table'
+  require 'open3'
+
+  def self.storage_report_text
+    head = %w[filesystem total used pct_used free pct_free]
+    Text::Table.new(head: head, rows: storage_report_lines).to_s
+  end
+
+  def workflow_report_text
+    head = %w[workflow waiting error recent archived]
+    Text::Table.new(head: head, rows: workflow_report_lines).to_s
+  end
+
+  def self.workflow_xml
+    File.open('config/workflows/sdr/preservationIngestWF') { |f| Nokogiri::XML(f) }
+  end
+
+  def self.repository
+    workflow_xml.at_xpath('//workflow-def/@repository').value
+  end
+
+  def self.ingest_wf
+    workflow_xml.at_xpath('//workflow-def/@id').value
+  end
+
+  def self.ingest_wf_steps
+    workflow_xml.xpath('//process/@name').map(&:value)
+  end
+
+  def self.df_output
+    stdout_str, _status = Open3.capture2('df -h')
+    stdout_str
+  end
+
+  def self.storage_mount_lines
+    df_output.split("\n").select { |line| line.include?('services-disk') }
+  end
+
+  def self.parsed_storage_mount_lines
+    storage_mount_lines.map do |line|
+      storage_root = {}
+      split_line = line.split(' ')
+      storage_root[:total] = split_line[0]
+      storage_root[:used] = split_line[1]
+      storage_root[:remaining] = split_line[2]
+      storage_root[:percent_used] = split_line[3]
+      storage_root[:filesystem] = split_line[4]
+      storage_root[:percent_free] = "#{(100 - split_line[3].chop.to_i)}%"
+      storage_root
+    end
+  end
+
+  def self.storage_report_lines
+    parsed_storage_mount_lines.map do |h|
+      [h[:filesystem], h[:total], h[:used], h[:percent_used], h[:remaining], h[:percent_free]]
+    end
+  end
+
+  private_class_method :storage_mount_lines, :parsed_storage_mount_lines, :storage_report_lines, :workflow_xml
+
+  private
+
+  def waiting_count
+    Dor::WorkflowService.count_objects_in_step(StatsReporter.ingest_wf, 'start-ingest',
+                                               'waiting', StatsReporter.repository)
+  end
+
+  def erroring_count
+    errors = 0
+    StatsReporter.ingest_wf_steps.each do |step|
```

Comment:
This is the equivalent of `map{...}.sum`, where the `errors` var is unnecessary.